addons/mail/static/src/discuss/discuss.js

```diff
@@ -67,7 +70,19 @@ export class Discuss extends Component {
         this.orm = useService("orm");
         this.effect = useService("effect");
         this.prevInboxCounter = this.store.discuss.inbox.counter;
-        useChildSubEnv({ inDiscussApp: true });
+        useChildSubEnv({
+            inDiscussApp: true,
+            messageHighlight: this.messageHighlight,
+            openPinMenu: () => {
+                this.state.activeMode = this.MODES.PINNED_MESSAGES;
+            },
+            closePinMenu: () => {
+                if (this.state.activeMode !== this.MODES.PINNED_MESSAGES) {
+                    return;
+                }
+                this.state.activeMode = this.MODES.NONE;
+            },
+        });
```

Comment:
Both functions work together. Would be best to pass `pinMenu: { open() {}, close() {} }` so a single object represent the feature to open/close the pin menu.